docs/customization.rst

```diff
@@ -0,0 +1,72 @@
+.. customization:
+
+Customization
+=============
+
+Often you need to modify certain aspects of Schemathesis behavior, adjust data generation, modify requests before
+sending and so on. Schemathesis offers a hook mechanism which is similar to the pytest's one.
+
+Hooks
+-----
+
+Basing on the changes scope there are three levels of hooks:
+
+- Global. These hooks applied to all schemas in the test run;
+- Schema-local. Applied only for specific schema instance;
+- Test function specific. Applied only for a specific test function;
+
+To register a new hook function you need to use special decorators - ``register`` for global and schema-local hooks and ``apply`` for test-specific ones:
+
+.. code:: python
+
+    import schemathesis
+
+    @schemathesis.hooks.register
+    def before_generate_query(strategy, context):
```

Comment:
Probably context should be passed as the first argument since it will be more obvious - it will always present in all hooks and on the first position